queue_services/entity-emailer/src/entity_emailer/message_tracker/tracker.py

```diff
@@ -171,16 +180,30 @@ def start_tracking_message(message_context_properties: dict, email_msg: dict, ex
     return create_processing_message(message_context_properties, email_msg)
 
 
-def complete_tracking_message(tracker_msg: MessageProcessing):
+def complete_tracking_message(tracker_msg: MessageProcessing, email_msg):
     """Update existing message state to COMPLETED."""
     update_message_status_to_complete(tracker_msg)
 
+    # Update corresponding furnishings entry as PROCESSED
+    etype = email_msg.get('type', None)
+    if etype and etype == 'bc.registry.dissolution':
+        furnishing_id = email_msg['data']['furnishing']['furnishingId']
+        ar_overdue_stage_1_notification.update_furnishing_status(furnishing_id,
+                                                                 Furnishing.FurnishingStatus.PROCESSED)
+
 
 def mark_tracking_message_as_failed(message_context_properties: dict,
                                     email_msg: dict,
                                     existing_tracker_msg: MessageProcessing,
                                     error_details: str):
     """Create a new message with FAILED status or update an existing message to FAILED status."""
+    # Update corresponding furnishings entry as FAILED
+    etype = email_msg.get('type', None)
+    if etype and etype == 'bc.registry.dissolution':
+        furnishing_id = email_msg['data']['furnishing']['furnishingId']
+        ar_overdue_stage_1_notification.update_furnishing_status(furnishing_id,
+                                                                 Furnishing.FurnishingStatus.FAILED)
+
```

Comment:
this should not be part of tracker code.  

should take place in processor

Reply:
I am not sure how to detect the HTTP status of notify API in the Processor. 😢
So, I implemented this feature in update_message_tracker

Reply:
I think what you can do is add a try/catch block around `send_email` in `worker.py`

![image](https://github.com/user-attachments/assets/99c1c2f9-8ea8-44ee-b075-ed0fb0380e49)

If there is no exception thrown, you can assume it sent the email successfully. 
 if an error is thrown then you can mark it as failed and re-throw the error so the original error handling logic can run.

In both instances, maybe a new function named something like `involuntary_dissolution_stage_1_processor.post_process` can be called

Reply:
Updated